test/blas_test.hpp

```diff
@@ -226,6 +229,29 @@ inline void dump_arg<float>(std::ostream &ss, float f) {
   }
 }
 
+template <>
```

Comment:
Is this change also related to the index type?

Reply:
No, these are related to `BLAS_DATA_TYPES`.

The test names are generated using the test parameters. However, a test's name can't contain `-` or `.`. This causes a problem for floating point test parameters. To circumvent this, the tests replace `-` for `m` and `.` for `p`. 

However, this fix has only been implemented for `float`. Consequently, before this PR, testing with `half` or `double` results in `-` or `.` in test names, causing the tests to fail. This changes ensures that a valid test names are generated when `BLAS_DATA_TYPES` includes `half` or `double`.

Reply:
The code in the template body is exact copy of float. I think there is a lof of code duplication here. You can abstract the code in the main template or put it in a struct so you can partially instantiate it for float, double or half. Also it is better to be in a separate PR as it is not related to the 64 index type.

Reply:
This change is removed in [010ab05](https://github.com/codeplaysoftware/sycl-blas/pull/339/commits/010ab05c690200c552f03736b8dc2afe3d412f05), and I've filed a new issue https://github.com/codeplaysoftware/sycl-blas/issues/340.